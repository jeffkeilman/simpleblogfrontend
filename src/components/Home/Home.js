import TitleBar from '../TitleBar/TitleBar'
import ArticleArea from '../ArticleArea/ArticleArea'

function Home () {
  return (
    <div className='Home'>
      <TitleBar />
      <ArticleArea />
    </div>
  )
}

export default Home
