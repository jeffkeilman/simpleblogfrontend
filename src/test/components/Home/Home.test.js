import React from 'react'
import renderer from 'react-test-renderer'
import Home from '../../../components/Home/Home'
import { render } from '@testing-library/react'

describe('Home', () => {
  // snapshot test
  it('renders correctly', () => {
    const home = renderer.create(<Home />).toJSON()
    expect(home).toMatchSnapshot()
  })

  it('renders TitleBar', () => {
    const { container } = render(<Home />)
    expect(container.querySelector('.TitleBar')).toBeInTheDocument()
  })

  it('renders ArticleArea', () => {
    const { container } = render(<Home />)
    expect(container.querySelector('.ArticleArea')).toBeInTheDocument()
  })
})
