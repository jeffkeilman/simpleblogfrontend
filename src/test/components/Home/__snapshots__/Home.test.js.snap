// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="Home"
>
  <header
    className="TitleBar"
  >
    <section
      className="InfoPanel"
    >
      <section
        className="Description"
      >
        <p>
          Hello. My name is 
          <span>
            Jeff Keilman
          </span>
          .
        </p>
        <p>
          I am a 
          <span>
            software engineer
          </span>
           with a passion for 
          <span>
            social justice
          </span>
          .
        </p>
        <p
          className="LinkParagraph"
        >
          Here are some of my 
          <span>
            other links
          </span>
          :
        </p>
        <a
          href="https://github.com/jeffkeilman"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub link"
            className="GitHubLink"
            src="GitHubMarkLogo.png"
          />
        </a>
        <a
          href="https://www.linkedin.com/in/jeffkeilman"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="LinkedIn link"
            className="LinkedInLink"
            src="LinkedInLogo.png"
          />
        </a>
        <p>
          And here is my 
          <span>
            blog
          </span>
          .
        </p>
      </section>
    </section>
    <section
      className="Face"
    >
      <img
        alt="Author's face"
        src="Face.jpg"
      />
    </section>
  </header>
  <section
    className="ArticleArea"
  >
    <article
      className="Article"
    >
      <sup>
        Posted on: 
        Nov 09, 2021
      </sup>
      <h1>
        My first article
      </h1>
      <p>
        This is my first article. I'm writing it to learn about blogging. I would like to become a more proficient writer. I don't anticipate this blah blah yadda yadda doesn't matter just need some more length
        ...
      </p>
    </article>
    <article
      className="Article"
    >
      <sup>
        Posted on: 
        Oct 21, 2021
      </sup>
      <h1>
        My second article
      </h1>
      <p>
        This is my second article. I'm writing it to learn about blogging. I would like to become a more proficient writer. I don't anticipate this blah blah yadda yadda doesn't matter just need some more length
        ...
      </p>
    </article>
    <article
      className="Article"
    >
      <sup>
        Posted on: 
        Oct 03, 2021
      </sup>
      <h1>
        My third article
      </h1>
      <p>
        This is my third article. I'm writing it to learn about blogging. I would like to become a more proficient writer. I don't anticipate this blah blah yadda yadda doesn't matter just need some more length
        ...
      </p>
    </article>
  </section>
</div>
`;
